ls1intum/Artemis
# `Communication`: Improve performance by reducing the notification size when creating messages

### Checklist
#### General


#### Server
- [x] I updated multiple integration tests (Spring) related to the features (with a high test coverage).
#### Client
no real client changes

### Motivation and Context
The payload for messages is too large and contains information unnecessary for the client.
We should send minimal payloads to prevent overloading the involved subsystems.

### Description
Hide details in the objects send to the client in the REST call response and when sending notifications via web sockets

### Steps for Testing

Prerequisites:
- 1 course (messaging enabled) and 2 Students

1. Open a channel side by side with 2 users
2. Open the Chrome console inspecting the REST call for user 1
3. Open the Chrome console inspecting WS message for user 2
4. Send a message with user 1
5. Make sure the REST response for user 1 contains minimal information: no exercise, no lecture, no exam, no course, no conversation participants
6. Make sure the WS messages for user 2 contain minimal information: no exercise, no lecture, no exam, no course, no conversation participants
7. Also test related cases like editing messages, reacting with emojis, deleting messages and other interactions
8. Make sure in all these cases no additional information is sent in WS messages nor REST responses


Due to refactoring, we need to test all other message types: replies/answers, one to one chats, group chats, etc.



### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
